scripts/check_compliance.py

```diff
@@ -211,6 +215,124 @@ def run(self):
 """.format(max_top_items, n_top_items, __file__)
 
 
+        #
+        # Look for references to undefined Kconfig symbols outside Kconfig
+        # files
+        #
+
+        # Grep for symbol references.
+        #
+        # Example output line for a reference to CONFIG_BAZ at line 17 of
+        # foo/bar.c:
+        #
+        #   foo/bar.c<null>17<null>CONFIG_BAZ
+        #
+        # Skip the samples/ and tests/ directories for now. They often contain
+        # Kconfig files that are not part of the main Kconfig tree, which will
+        # trigger false positives until we do something fancier. Skip
+        # doc/releases too, which often references removed symbols.
+        grep_cmd = "git grep --only-matching --line-number -I --null " \
+                   "--extended-regexp --word-regexp CONFIG_[A-Z0-9_]+ " \
+                   "-- :!samples :!tests :!doc/releases"
+
+        grep_process = subprocess.Popen(grep_cmd.split(),
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE,
+                                        cwd=self.zephyr_base)
+
+        grep_stdout, grep_stderr = grep_process.communicate()
+        if grep_process.returncode:
+            self.case.result = Error("error running 'git grep'", "error")
+            # grep_stderr will be 'bytes' here, which prints as b'oops'
+            self.case.result._elem.text = \
+                "'{}' failed with exit code {} and wrote {} to stderr " \
+                "(while searching for Kconfig symbol references)" \
+                .format(grep_cmd, grep_process.returncode, grep_stderr)
```

Comment:
Seems fine to me. One issue is that `stderr` was added to `CalledProcessError` in Python 3.5. I think we still support Python 3.4, at least on paper. @nashif?

Looking at [the documentation](https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError), it seems `check_output()` might not capture `stderr` either, but only `run()` (which was added in Python 3.5).

Reply:
Hmm... I guess the `stderr=subprocess.STDOUT` might make it work still. Feels a bit gnarly to include `stderr` in the parsed output though, even if it'd probably always be empty in practice.

Reply:
> > This is dropping grep_stdout in case of error.

> Feels a bit gnarly to include stderr in the parsed output though

You just helped me realize the current code also drops grep_stderr in case of success, think warnings. Maybe merging stderr with stdout and breaking the stdout parser on purpose is actually a good thing? It would mean "warnings" aren't silently discarded anymore.

Another, .communicate() alternative is to treat warnings as errors with:

```python
if grep_process.returncode or grep_stderr:
   Error(...
```

PS: this is exactly why people still use shell scripts for simple tasks (_not_ this one)
For convenience: https://docs.python.org/3.4/library/subprocess.html

Reply:
I did the `or grep_stderr` thing.